controllers/operatorpolicy_controller.go

```diff
@@ -329,7 +352,79 @@ func (r *OperatorPolicyReconciler) buildResources(policy *policyv1beta1.Operator
 		}
 	}
 
-	return sub, opGroup, updateStatus(policy, validationCond(validationErrors)), nil
+	return sub, opGroup, updateStatus(policy, validationCond(validationErrors)), returnedErr
+}
+
+// applySubscriptionDefaults will set the subscription channel, source, and sourceNamespace when they are unset by
+// utilizing the PackageManifest API.
+func (r *OperatorPolicyReconciler) applySubscriptionDefaults(
+	ctx context.Context, subscription *operatorv1alpha1.Subscription,
+) error {
+	subSpec := subscription.Spec
+
+	defaultsNeeded := subSpec.Channel == "" || subSpec.CatalogSource == "" || subSpec.CatalogSourceNamespace == ""
+
+	if !defaultsNeeded {
+		return nil
+	}
+
+	// PackageManifests come from an API server and not a Kubernetes resource, so the DynamicWatcher can't be used since
+	// it utilizes watches. The namespace doesn't have any meaning but is required.
+	packageManifest, err := r.DynamicClient.Resource(packageManifestGVR).Namespace("default").Get(
+		ctx, subSpec.Package, metav1.GetOptions{},
+	)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return fmt.Errorf(
+				"%wthe subscription defaults could not be determined because the PackageManifest was not found",
+				ErrPackageManifest,
+			)
+		}
+
+		log.Error(err, "Failed to get the PackageManifest", "name", subSpec.Package)
+
+		return fmt.Errorf(
+			"%wthe subscription defaults could not be determined because the PackageManifest API returned an error",
+			ErrPackageManifest,
+		)
+	}
+
+	catalog, _, _ := unstructured.NestedString(packageManifest.Object, "status", "catalogSource")
+	catalogNamespace, _, _ := unstructured.NestedString(packageManifest.Object, "status", "catalogSourceNamespace")
+
+	if catalog == "" || catalogNamespace == "" {
+		return fmt.Errorf(
+			"%wthe subscription defaults could not be determined because the PackageManifest didn't specify a catalog",
+			ErrPackageManifest,
+		)
+	}
```

Comment:
I'm not sure that this one should wrap the error so it's retried - I think usually this would get fixed by the content in the catalog changing? What was your reasoning

Reply:
I don't know that this would ever occur, but I figured it was likely due to partial results being returned but it seems unlikely. I'll remove the retry.